libcontainer/nsenter/nsexec.c

```diff
@@ -164,11 +164,16 @@ static void write_log(const char *level, const char *format, ...)
 	if (ret < 0)
 		goto out;
 
-	dprintf(logfd, "{\"level\":\"%s\", \"msg\": \"%s[%d]: %s\"}\n", level, stage, getpid(), message);
+	ret = asprintf(&json, "{\"level\":\"%s\", \"msg\": \"%s[%d]: %s\"}\n", level, stage, getpid(), message);
+	if (ret < 0)
```

Comment:
`json` has to be NULLified on error (or avoid `free` on error)

https://man7.org/linux/man-pages/man3/asprintf.3.html

Reply:
Indeed! Fixed.